client/my-sites/email/email-management/home/utils.js

```diff
@@ -98,11 +99,23 @@ export function hasEmailSubscription( domain ) {
 	return !! subscriptionId;
 }
 
-export function resolveEmailPlanStatus( domain ) {
+/**
+ * Determines if any warnings exists with the slug `unused_mailboxes` in an array of warning objects
+ *
+ * @param {Array} warnings - An array of warning objects
+ * @returns {boolean} true if unused mailboxes exists, false otherwise
+ */
+function hasUnusedMailboxWarnings( warnings ) {
```

Comment:
Two thoughts:
1. Does it make more sense to pass the `account` object to this helper function?
2. Might this function be more appropriate in `client/lib/emails`?

Reply:
> Does it make more sense to pass the account object to this helper function?

Right now it has all the data it needs IMO. Also consider that the logic is already gated by features like `hasTitanMailWithUs` or `hasGSuiteWithUs` for this use case. That particular slug works the same way for both plan types and that function only needs to check for the presence of the warning slug. If this changes though,  I'll be glad to refactor it.

> Might this function be more appropriate in client/lib/emails?

It's only used once ATM. It should be easy to refactor that to the `client/lib/emails` location when we need to.

Reply:
Fair enough. I've been working on similar functions and have them all in `client/lib/emails` in my upcoming PR, so we can hash the details out in that PR.